transit_odp/organisation/csv/overall.py

```diff
@@ -116,6 +170,24 @@ def _get_overall_catalogue_dataframe() -> DataFrame:
         for old_name, column_tuple in OVERALL_COLUMN_MAP.items()
     }
     merged = merged[OVERALL_COLUMN_MAP.keys()].rename(columns=rename_map)
+
+    df_data_id_list = merged["Data ID"].tolist()
+    all_data_dict = get_fares_data()
+
+    for data_id in df_data_id_list:
```

Comment:
This is very much an antipattern and slow. Pandas dataframes are not meant to be iterated over like this

Reply:
@ITO-PaulHolland - Do you mean that I shouldn't iterate through a single column, instead iterate through the rows instead?

Reply:
I meant doing a join or a concat or something using the pandas API.